extensions/wikia/SpecialMultipleLookup/ApiQueryMultiLookup.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+/**
+ * Provides API module to look up activity from a given IP address across all of FANDOM
+ */
+class ApiQueryMultiLookup extends ApiQueryBase {
+	/** @var DatabaseBase $db */
+	private $db;
+
+	public function __construct( ApiBase $query, $moduleName ) {
+		parent::__construct( $query, $moduleName, 'ml' );
+	}
+
+	/**
+	 * @inheritdoc
+	 */
+	public function execute() {
+		if ( !$this->getUser()->isAllowed( 'multilookup' ) ) {
+			$this->dieUsageMsg( 'badaccess-groups' );
+		}
+
+		$params = $this->extractRequestParams();
+
+		if ( !IP::isIPAddress( $params['target'] ) ) {
+			$this->dieUsage( 'Invalid IP', 'brokenip', 400 );
+		}
+
+		$ipAddress = IP::sanitizeIP( $params['target'] );
+
+		$this->addTables( 'multilookup' );
+		$this->addFields( '*' );
+		$this->addWhereFld( 'ml_ip_bin', inet_pton( $ipAddress ) );
+
+		if ( !empty( $params['offset'] ) ) {
+			$ts = $this->getDB()->timestamp( $params['offset'] );
+			$this->addWhere( 'ml_ts < ' . $this->getDB()->addQuotes( $ts ) );
+		}
+
+		$this->addOption( 'ORDER BY', 'ml_ts DESC' );
+
+		$result = $this->getResult();
+
+		foreach ( $this->select( __METHOD__ ) as $row ) {
+			$wiki = WikiFactory::getWikiByID( $row->ml_city_id );
+
+			if ( !$wiki ) {
+				continue;
+			}
+
+			foreach ( $this->getActivityOnWiki( $wiki->city_dbname, $ipAddress ) as $user ) {
+				$activity = [
+					'wiki' => $wiki->city_url,
+					'user' => $user,
+					'date' => $row->ml_ts
+				];
+
+				if ( !$result->addValue( null, null, $activity ) ) {
+					$this->setContinueEnumParameter( 'offset', $row->ml_ts );
+				}
+			}
+		}
+
+		$result->setIndexedTagName_internal( null, 'mlactivity' );
+	}
+
+	/**
+	 * @inheritdoc
+	 */
+	protected function getDB() {
+		if ( empty( $this->db ) ) {
+			global $wgSpecialsDB;
+			$this->db = wfGetDB( DB_SLAVE , 'api', $wgSpecialsDB );
+		}
+
+		return $this->db;
+	}
+
+	private function getActivityOnWiki( $dbName, $ipAddress ) {
+		$cacheKey = wfSharedMemcKey( 'multilookup', $this->target, $dbName );
```

Comment:
Shouldn't we use `$ipAddress` instead of `$this->target` (which I do not see being set anywhere)?